### ISSUE TITLE
Parallel sources downloads

### ISSUE THREAD

**Issue by je-vv** [AladW/aurutils#777]
Hi !
I understand `aur sync` can parallelize getting the AUR packages through `AUR_QUERY_PARALLEL=1`, however once all AUR packages are downloaded, it goes and processes one by one.  For the build, install and such parts of the PKGBUILD, that's fine, and actually I wouldn't want several packages building at the same time, given I'm using `MAKEFLAGS="-j$(nproc --ignore 1)"` in "/etc/makepkg.conf" already.
However, when fetching all sources specified on `source`, `source_<arch>`, and corresponding signature files if specified, for all PKGBUILDs being synced, I see doing one by one being inconvenient.  I'm wondering if all those fetches/downloads could be done in parallel as well.

**Comment by Earnestly** [comment]
This is a `makepkg` question and so likely out of scope for aur utils.  I don't believe `makepkg` has a way to download things in parallel outside of setting custom `DLAGENTS`.  See [**makepkg.conf**(5)](https://jlk.fjfi.cvut.cz/arch/manpages/man/makepkg.conf.5.en)

**Comment by AladW** [comment]
AUR packages/repositories are still retrieved sequentially - 
AUR_QUERY_PARALLEL only sets if AUR _requests_ (that is, requests to the 
API for dependencies and other information) are done in parallel.

In aurutils 1.5 there was an option to retrieve repositories in 
parallel, but it only worked well for tar snapshots, see e.g. commit 
5e2b0c3f243e5f4131da02f52ea513fe93087d98.

Downloading the actual sources in parallel is actually not too 
difficult: you run `env -C <package directory> makepkg --verifysource 
-o` in parallel. It's not difficult to implement this by yourself with 
aur sync -o:

#!/bin/sh
aur sync -o "$@" | while read -r path; do
   env -C "$path" makepkg --verifysource -o &
done
aur sync --continue "$@"

You might limit the amount of background jobs with `wait -n`, see 
https://github.com/AladW/aurutils/blob/master/lib/aur-srcver#L73-L85.

Javier wrote:

**Comment by je-vv** [comment]
Hi @AladW !  Looking at your last answer, that sounds like it, :)  Though not sure if would be wise to incorporate that into aur-sync eventually...  And I misunderstood the AUR_QUERY_PARALLEL then, :(  It would have been nice to download the AUR packages in parallel as well, but I guess as they're small in size it doesn't matter much, the packages sources would be the ones benefiting the most...
Thanks a lot !
